digdag-docs/src/operators/py.md

```diff
@@ -76,6 +76,111 @@ See [Python API documents](../python_api.html) for details including variable ma
     print("simple execution")
   ```
 
+  You can pass arguments to class for initialization　by defining arguments under the `py>:` operation as the following:
+  ```yaml
+  # sample.dig
+  +some_task:
+    py>: tasks.MyWorkflow.my_task
+    required1_1: awesome execution
+    required1_2: "awesome execution"
+    required2: {a: "a"}
+    required3: 1
+    required4: 1.0
+    required5: [a, 1, 1.0, "a"]
+  ```
+
+  Also, you can do the same thing using `_export` as the following:
+  ```yaml
+  # sample.dig
+  +some_task:
+    _export:
+      required1_1: awesome execution
+      required1_2: "awesome execution"
+      required2: {a: "a"}
+      required3: 1
+      required4: 1.0
+      required5: [a, 1, 1.0, "a"]
+    py>: tasks.MyWorkflow.my_task
+  ```
+
+  This example assume following Python script:
+
+  ```python
+  # tasks.py
+  class MyWorkflow(object):
+      def __init__(
+        self,
+        required1_1: str,
+        required1_2: str,
+        required2: dict[str, str],
+        required3: int,
+        required4: float,
+        required5: list[Union[str, int, float]]
+      ):
+          print(f"{required1_1} same as {required1_2}")
+          self.arg2 = required2
+          print(f"{float(required3)} same as {required4}")
+          self.arg5 = required5
+      
+      def my_task(self):
+          pass
+  ```
+
+  Or, you can pass arguments to function as the following:
+
+  ```yaml
+  # sample.dig
+  +some_task:
+    py>: simple_tasks.my_func
+    required1: simple execution
+    required2: {a: "a"}
+  ```
+
+  ```yaml
+  # simple_sample.dig
+  +some_task:
+    _export:
+      required1: simple execution
+      required2: {a: "a"}
+    py>: simple_tasks.my_func
+  ```
+
+  ```python
+  # simple_tasks.py
+  def my_func(required1: str, required2: dict[str, str]):
+    print(f"{required1}: {required2}")
+  ```
+
+  Finally, you can pass combination of class and mehtod arguments to Python script as the following:
```

Comment:
This behavior is a bit strange. 
IIUC, we can't use same argument name between class initializer and method argument for it.
It might be better to add a warning about it.

Reply:
Sorry for my poor English. Just to be clear, what I tried to explain is in digdag, indeed we can pass both(class and method) arguments at once.
So, on-premise that the above behavior is correct what I understood is I just need to add a warning like ‘But you can not pass combination with the same name.’ instead of modifying the example.
Is this correct? @yoyama

Reply:
@magicpieh28 Sorry for my comment is not clear.
Yes, your description and example are correct. But I feel a bit strange on this specification on `py>`.
Especially, if both initializer of Class and the method have same parameter name, we can't set them separately. (If I understand correctly).
So I propose add description to warn about it as you describe.
Anyway thank you for your contribution.